haskell_lint errors lost verbosity

#258 reduced the verbosity of the linter errors, removing the specific error description.

The following is the output of running the rule `haskell_lint(name = "lint", deps = ["test"])`, where the library `test` contains one file with an unused top bind.

In version 93a8b235006123171ab831d05d85bf30e38021f5:
```
$ bazel build lint
/nix/store/qj9hgyyf41blhl30f5n3kyww7nlb0sif-numero-toolchain
INFO: Analysed target //:lint (6 packages loaded).
INFO: Found 1 target...
ERROR: /home/z/src/tweag/bazel-test/BUILD:19:1: Linting test failed (Exit 1)

Test.hs:4:1: warning: [-Wunused-top-binds]
    Defined but not used: `aoe'
  |
4 | aoe = 8
  | ^^^

<no location info>: error:
Failing due to -Werror.
Target //:lint failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 5.415s, Critical Path: 0.59s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

In version 3def6540cdcf866cfe22d8bd460d8af80f7f7c11:
```
$ bazel build lint
INFO: Analysed target //:lint (0 packages loaded).
INFO: Found 1 target...
ERROR: /home/z/src/tweag/bazel-test/BUILD:19:1: error executing shell command: '
    bazel-out/k8-fastbuild/bin/visible-binaries/ghc "$@" > bazel-out/k8-fastbuild/bin/lint-log-test-1.0.0 2>&1
    ' failed (Exit 1)
Target //:lint failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.505s, Critical Path: 0.24s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

Bazel version: `0.13.0- (@non-git)`

--- a/haskell/lint.bzl
+++ b/haskell/lint.bzl
@@ -90,17 +90,21 @@ def _haskell_lint_aspect_impl(target, ctx):
       set.to_depset(build_info.interface_files),
       set.to_depset(build_info.dynamic_libraries),
       depset(build_info.external_libraries.values()),
-      depset([hs.tools.ghc]),
+      depset([
+        hs.tools.ghc,
+        hs.tools.cat,
+      ]),
     ]),
     outputs = [lint_log],
     progress_message = "Linting {0}".format(ctx.rule.attr.name),
     command = """
-    {ghc} "$@" > {output} 2>&1
+    (ghc "$@" > {output} 2>&1) || (last_code=$? && cat {output} && exit $last_code)
     """.format(
       ghc = tools(ctx).ghc.path,
       output = lint_log.path,
     ),
     arguments = [args],
+    env = hs.env,
   )
 
   return [HaskellLintInfo(
@@ -187,17 +191,18 @@ def _haskell_doctest_aspect_impl(target, ctx):
       set.to_depset(build_info.interface_files),
       set.to_depset(build_info.dynamic_libraries),
       depset(build_info.external_libraries.values()),
-      depset([tools(ctx).doctest]),
+      depset([
+        hs.tools.doctest,
+        hs.tools.cat,
+      ]),
     ]),
     outputs = [lint_log],
     progress_message = "Doctesting {0}".format(ctx.rule.attr.name),
     command = """
-    doctest "$@" > {output} 2>&1
+    (doctest "$@" > {output} 2>&1) || (last_code=$? && cat {output} && exit $last_code)
     """.format(output = lint_log.path),
     arguments = [args],
-    env = {
-      "PATH": get_build_tools_path(ctx),
-    },
+    env = hs.env,
   )
 
   return [HaskellLintInfo(

--- a/haskell/lint.bzl
+++ b/haskell/lint.bzl
@@ -98,9 +98,9 @@ def _haskell_lint_aspect_impl(target, ctx):
     outputs = [lint_log],
     progress_message = "Linting {0}".format(ctx.rule.attr.name),
     command = """
-    (ghc "$@" > {output} 2>&1) || (last_code=$? && cat {output} && exit $last_code)
+    ghc "$@" > {output} 2>&1 || rc=$? && cat {output} && exit $rc
     """.format(
-      ghc = tools(ctx).ghc.path,
+      ghc = hs.tools.ghc.path,
       output = lint_log.path,
     ),
     arguments = [args],

--- a/haskell/toolchain.bzl
+++ b/haskell/toolchain.bzl
@@ -137,9 +137,10 @@ def _haskell_toolchain_impl(ctx):
 
   targets_w = [
     "bash",
-    "ln",
+    "cat",
     "cp",
     "grep",
+    "ln",
     "mkdir",
   ]
 

